tvOS 14 Fix

nicksaintjohn opened the pull request:
Not perfect, but it gets things running again. Simply disables VisualEffectView tweaks that are causing the crash.

Thanks to petrusje, for this fix.

yvan-sraka committed:
Add macOS Catalyst support

(cherry picked from commit 026c054cfa1b3801e52e6715685e1f8faab47da0)

Commit:
Improve side menu style

nicksaintjohn committed:
tvOS 14 Fixes. Not perfect, but it gets things running again.

Thanks to petrusje, for this fix.

iAYMANi commented:
.ipa

rsemedo2002 commented:
Can someone share .ipa file, please...?

Midifreakz commented:
Can you please share the .ipa file? much appreciated!

nicksaintjohn commented:
I'm not sure if should post the .ipa, whilst using my own team and signing certificate? I've been looking for ways to do so without signing, but have reached a dead end.

Should I be changing the bundle ID? Is it possible to compile an unsigned version? I'm unsure of the best practice here. Is anyone able to explain the common practice? Thanks

Midifreakz commented:
I have just compiled my own version but I'm stuck at the .app to .ipa conversion... Can anyone point me in the right direction?

Midifreakz commented:
@nicksaintjohn you can use a signing service like http://signulous.com. This replaces your signing certificate with theirs. should be okay to go. It is a paid service but if you want I can resign it for you

nicksaintjohn commented:
> @nicksaintjohn you can use a signing service like http://signulous.com. This replaces your signing certificate with theirs. should be okay to go. It is a paid service but if you want I can resign it for you

Is there any risk, or breach of rules to post an .ipa that is self-signed? I presume everyone has to re-sign it anyhow, before side-loading to their own device?

Perhaps there's some kind of vanilla signing certificate available, that I could use in conjunction with 'iOS App Signer' to get a similar result?

I'm happy to PM you a build, if you would kindly re-sign it... How best to get it to you?

hussainmahm commented:
@nicksaintjohn Perhaps this video can help you: https://www.youtube.com/watch?v=SXRimW23AjI

nicksaintjohn commented:
> @nicksaintjohn Perhaps this video can help you: https://www.youtube.com/watch?v=SXRimW23AjI

Sadly, I've already spend a few hours trying out these methods, hoping to get it to compile without using a certificate, but with so many targets I guess it's too easy to miss something... I keep getting signing errors. Have also tried with command line, to no avail.

Midifreakz commented:
Hi Nick, I was trying to figure out a way to get it to me without exposing too much information. I made this dummy e-mail adres: nicotaxibackups@gmail.com you can send it to me there and I we can discuss this over e-mail if you like? I will remove this post when I have received your mail.

nicksaintjohn commented:
> Hi Nick, I was trying to figure out a way to get it to me without exposing too much information. I made this dummy e-mail adres: [nicotaxibackups@gmail.com](mailto:nicotaxibackups@gmail.com) you can send it to me there and I we can discuss this over e-mail if you like? I will remove this post when I have received your mail.

Have emailed you 👍

Midifreakz commented:
Hi Nick, could you remove your reply with the e-mail adres still visible? haha

robertovg commented:
Yes, could be fine just to get an un-signed IPA to sign with http://signulous.com and don't deal with XCode specific build issues...

Thanks in advance.

nicksaintjohn commented:
If anyone would like to try this .ipa, it will need re-signing, please let me know how you get on!

https://we.tl/t-L6TM0AtubQ

Not sure of best practices here, so haven't posted the .ipa directly. Perhaps someone can let me know what's normal. Thanks

barry74 commented:
> If anyone would like to try this .ipa, it will need re-signing, please let me know how you get on!
> 
> https://we.tl/t-L6TM0AtubQ
> 
> Not sure of best practices here, so haven't posted the .ipa directly. Perhaps someone can let me know what's normal. Thanks

Working great!
Many thanks!!

stevo-t commented:
> > If anyone would like to try this .ipa, it will need re-signing, please let me know how you get on!
> > https://we.tl/t-L6TM0AtubQ
> > Not sure of best practices here, so haven't posted the .ipa directly. Perhaps someone can let me know what's normal. Thanks
> 
> Working great!
> Many thanks!!



> > If anyone would like to try this .ipa, it will need re-signing, please let me know how you get on!
> > https://we.tl/t-L6TM0AtubQ
> > Not sure of best practices here, so haven't posted the .ipa directly. Perhaps someone can let me know what's normal. Thanks
> 
> Working great!
> Many thanks!!

I tested and it's no longer crashing.
Thanks!!!

robertovg commented:
Same here, it works like a charm

On Mon, 21 Sep 2020 at 21:26, stevo-t <notifications@github.com> wrote:

>
>
>
>
>
>
> If anyone would like to try this .ipa, it will need re-signing, please let
> me know how you get on!
>
>
> https://we.tl/t-L6TM0AtubQ
>
>
> Not sure of best practices here, so haven't posted the .ipa directly.
> Perhaps someone can let me know what's normal. Thanks
>
>
>
>
> Working great!
>
>
> Many thanks!!
>
>
>
>
>
>
>
>
> If anyone would like to try this .ipa, it will need re-signing, please let
> me know how you get on!
>
>
> https://we.tl/t-L6TM0AtubQ
>
>
> Not sure of best practices here, so haven't posted the .ipa directly.
> Perhaps someone can let me know what's normal. Thanks
>
>
>
>
> Working great!
>
>
> Many thanks!!
>
>
>
>
> I tested and it's no longer crashing.
>
>
> Thanks!!!
>
>
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/PopcornTimeTV/PopcornTimeTV/pull/929#issuecomment-696322230>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAELKN4JWQKEFEXCCHGE3HDSG6SF3ANCNFSM4RQMD4ZA>
> .
>
>
> --

-- 
Roberto Vázquez González

barry74 commented:
The found movies or shows below the Search area can't be selected with the Siri Remote.
The remote-app on iPhone works fine, you can press the return/go button and select.

Thanks for all the good work!

hussainmahm commented:
> The found movies or shows below the Search area can't be selected with the Siri Remote.
> The remote-app on iPhone works fine, you can press the return/go button and select.
> 
> Thanks for all the good work!

Have you tried restarting the Apple TV? If it works with the remote app then I'd assume it's not an issue with Popcorn.

jackazz86 commented:
I confirm what barry74 said: the found movies in the search area can’t be selected. I tried restarting my ATV4K but still had no luck. I’m not even able to select them with the remote app. Only way is to add titles via the watchlist on iPhone or other devices (on iPad works fine).

Also, oddly: if the research shows more than one title you are able to swipe down and select a title. If the research shows only one title you can’t swipe down, no way.

Thank you very much for making the app working again in iOS14!

barry74 commented:
>I’m not even able to select them with the remote app.

Yes, in remote-app tap in Text Area. Then return/go on keyboard.

jackazz86 commented:
This way it works! Thanks everybody.

Bonovox78 commented:
I’m getting the “Unknown error” message under Movies/tv shows. “A server with the specified hostname could not be found”.

hussainmahm commented:
What version does it show in settings? 3.2.36? I built mine from the latest code, did the fix and compiled it. No issues with search or unknown error.

Bonovox78 commented:
Apologies, I was using an old version under the same .ipa file as yours. I deleted the old one signed the new one and it works. Thanks.

Bonovox78 commented:
Anyone got an ipa for iOS 14 for Popcorn Time that works post iOS 14? Signulous have removed theirs.

Midifreakz commented:
@Bonovox78  You can download the one shared by the fork operator Nicksaintjohn. https://wetransfer.com/downloads/37d178339d87de55ba2a9077a254baa220200920160838/19d7eb

You will have to "resign" it with a service like Signulous.

Midifreakz commented:
@nicksaintjohn I noticed that the Wetransfer link you provided expires in 5 days... Maybe better host it somewhere else?

Bonovox78 commented:
@Midifreakz I meant the iOS version. I have the tvOS version.

Midifreakz commented:
Aha! If i'm not mistaking there was someone on the original fork of the PT project that made the iOS version .ipa available.

rsemedo2002 commented:
All working, thank you for the fantastic work! ✌️

nicksaintjohn commented:
> The found movies or shows below the Search area can't be selected with the Siri Remote.
> The remote-app on iPhone works fine, you can press the return/go button and select.
> 
> Thanks for all the good work!

I'm not seeing this behaviour at all, so I think it may be limited to your configuration. Hope you can solve it.

hussainmahm commented:
@nicksaintjohn can you try to search for something that only gives you 1 or 2 results and then try to scroll down? I'm seeing this behavior as well, if it's 3 or more results it would let me scroll down from Movies down to the results, but if there's only up to 2 results it won't scroll down past Movies or Shows or People. Another thing is once you scroll down to the results, you can't scroll back up to the search field, it just goes to the top navigation.

nicksaintjohn commented:
> @nicksaintjohn can you try to search for something that only gives you 1 or 2 results and then try to scroll down? I'm seeing this behavior as well, if it's 3 or more results it would let me scroll down from Movies down to the results, but if there's only up to 2 results it won't scroll down past Movies or Shows or People. Another thing is once you scroll down to the results, you can't scroll back up to the search field, it just goes to the top navigation.

Ah, yes, I can confirm the issue! Whilst I'm not sure what's causing it, I would expect it's due to the new keyboard and search api in tvOS14. Annoyingly, the best fixes for now are to search less specifically, then to restart the app once you've used up your single search... I'll post a updated version if I manage to find a fix for it 👍

Monz-TV commented:
@nicksaintjohn  you are the best working perfectlty but there is issue in search I cant scroll down 

,cant wait for the update to fix the search issue ,

Monz-TV commented:
> > @nicksaintjohn can you try to search for something that only gives you 1 or 2 results and then try to scroll down? I'm seeing this behavior as well, if it's 3 or more results it would let me scroll down from Movies down to the results, but if there's only up to 2 results it won't scroll down past Movies or Shows or People. Another thing is once you scroll down to the results, you can't scroll back up to the search field, it just goes to the top navigation.
> 
> Ah, yes, I can confirm the issue! Whilst I'm not sure what's causing it, I would expect it's due to the new keyboard and search api in tvOS14. Annoyingly, the best fixes for now are to search less specifically, then to restart the app once you've used up your single search... I'll post a updated version if I manage to find a fix for it 👍

i tried but am not able to select it even if its 1 result

hussainmahm commented:
@ Monz-TV you should be able to select results if there’s 3 or more results.

nicksaintjohn commented:
Refreshed link to a compiled .ipa (needs re-signing - which can be done using Signulous)

https://we.tl/t-J8jcnpJYGo

thatbroom commented:
@nicksaintjohn Is there any chance you could resupply the ipa link as the one from 8 days ago is now down, it would be really appreciated. Cheers

Ahmedxway commented:
> Refreshed link to a compiled .ipa (needs re-signing - which can be done using Signulous)
> 
> https://we.tl/t-J8jcnpJYGo

Can you please upload the link again as it’s expired now. Thanks alot :))

Bonovox78 commented:
Just wondering any fix for the issue of not being able to select when 1 show is shown in the results?

nicksaintjohn commented:
Hi guys, refreshed link to the compiled .ipa - It will need re-signing, which can be done easily using a service such as Signulous

https://we.tl/t-J4EFaQPST3

Sadly, no fix to the search issue yet

Midifreakz commented:
Hi Nick, Did you notice that movies are not updating anymore? New movies are not listed.. Something with the API perhaps?

nicksaintjohn commented:
> Hi Nick, Did you notice that movies are not updating anymore? New movies are not listed.. Something with the API perhaps?

I hadn't. When selecting 'newest', I saw some new random stuff that I didn't recognise, so assumed it was just the 'Covid effect' reducing big new releases. I suspect there's no problem with the app, and it's the tracker that is either not updating, or caching somehow, so likely the API, as you suggested... May investigate further when I get a chance.

nicksaintjohn committed:
Gets the app running again (crashing due to missing bluetooth auth check)
Currently can only play external torrents, or completed downloads (avoiding quality selection)
Disabling quality selector doesn't seem to help (as seen previously with similar issue)
Suspect UI issue due to segue changes. Have tried switching loader to modal, but still no dice

maussie1811 commented:
> Hi guys, refreshed link to the compiled .ipa - It will need re-signing, which can be done easily using a service such as Signulous
> 
> https://we.tl/t-J4EFaQPST3
> 
> Sadly, no fix to the search issue yet

Many Thanks it works great! for me at tvOS 14.0.1 Thanks!

asmith2986 commented:
can you please repost the link

asmith2986 commented:
> > Hi guys, refreshed link to the compiled .ipa - It will need re-signing, which can be done easily using a service such as Signulous
> > https://we.tl/t-J4EFaQPST3
> > Sadly, no fix to the search issue yet
> 
> Many Thanks it works great! for me at tvOS 14.0.1 Thanks!

can you please refresh link. thank you!

maussie1811 commented:
> > > Hi guys, refreshed link to the compiled .ipa - It will need re-signing, which can be done easily using a service such as Signulous
> > > https://we.tl/t-J4EFaQPST3
> > > Sadly, no fix to the search issue yet
> > 
> > 
> > Many Thanks it works great! for me at tvOS 14.0.1 Thanks!
> 
> can you please refresh link. thank you!

Here is a new link for a week. https://fil.email/GpzkjiPm

Only what to do is resign it yourself via a service like signulous or With your own paid Apple developer account.

Enjoy!

nicksaintjohn committed:
Merge branch 'tvOS-14-fix'

nicksaintjohn committed:
Merge remote-tracking branch 'refs/remotes/menu-fix/improve-side-menu-style'

nicksaintjohn committed:
Switched to using a third-party scraper, as the community one seems to be borked

nicksaintjohn committed:
Minor tweaks to mitigate the reported 'search bug' - No idea what's causing it, but have been able to bypass by using a combination of hacks. This needs fixing properly by someone with a better understanding of the root cause. This patch will result in a minimum of three results (when at least one item is found), by generating dummy objects when necessary.

nicksaintjohn committed:
Added UIVisualEffectView behind the media description for DescriptionViewController - Makes the detail text easier to read

nicksaintjohn committed:
Trakt fixes to prevent crashes for erroneous media with missing id's
Additional fix prevents a crash resulting from running TraktManager on the background thread, instead of main

nicksaintjohn committed:
Another fix for Trakt issues, whereby media items with no id will cause an instant crash when appearing onDeck - Not sure if this is a good fix, but it stops the immediate crashing. Could do with testing if there are any remaining issues

nicksaintjohn committed:
Various minor tweaks to reduce crashes and freezes

nicksaintjohn committed:
Quick and hacky switch to pctapi. It's pretty buggy currently, but it's a start. Too many fundamental changes may lead to this branch being abandoned, as the new scraper is structured quite differently, and may require a few things to be entirely re-written..

nicksaintjohn committed:
Fixes to "old-version" PCTAPI requests (may require ongoing tweaking)
Added APPLICATION_EXTENSION_API_ONLY to allow compile on XC >= v12.5
Restricted quality of listed torrents (duplicates still shown)
